html/mycakeapp/src/Model/Validation/CustomValidation.php

```diff
@@ -17,4 +17,27 @@ public function HalfSizeAlphanumericOnly($value)
     {
         return (bool)preg_match(" /^[a-zA-Z0-9]+$/", $value);
     }
+    // 半角英字と半角のスペースのみ許可
+    public function HalfSizeAlphabetAndSpaceOnly($value)
+    {
+        return (bool)preg_match(" /^[a-zA-Z ]+$/", $value);
+    }
+    public function IsDeadline($value)
+    {
+        $isDateFormat = (bool)preg_match(" /^[0-9]{2}\/[0-9]{2}$/", $value);
+        if ($isDateFormat === false) {
+            return false;
+        }
+        $thisYear = (int)date('y');
+        $thisMonth = (int)date('m');
+        $month = (int)substr($value, 0, 2);
+        $year = (int)substr($value, 3, 2);
+        if ($month > 12 || $year < $thisYear) {
+            return false;
+        }
+        if ($month <= $thisMonth && $year === $thisYear) {
```

Comment:
上と同じ理由です。今月分の登録は許可した方がいいと思います！
```suggestion
        if ($month < $thisMonth && $year === $thisYear) {
```